xla/service/gpu/fused_mha_thunk.cc

```diff
@@ -169,8 +175,21 @@ Status FusedMHABackwardThunk::ExecuteOnStream(const ExecuteParams& params) {
   se::DeviceMemoryBase d_bmm2_rhs_buffer =
       buffer_allocations.GetDeviceAddress(d_bmm2_rhs_buffer_);
 
-  se::DeviceMemoryBase d_S_buffer =
-      buffer_allocations.GetDeviceAddress(d_s_buffer_);
+  se::DeviceMemoryBase d_s_buffer;
+  if (d_s_buffer_.allocation() != nullptr) {
+    d_s_buffer = buffer_allocations.GetDeviceAddress(d_s_buffer_);
+  }
+
+  se::DeviceMemoryBase softmax_sum_buffer;
+  if (softmax_sum_buffer_.allocation() != nullptr) {
+    softmax_sum_buffer =
+        buffer_allocations.GetDeviceAddress(softmax_sum_buffer_);
+  }
+
+  se::DeviceMemoryBase d_Q_accum_buffer;
+  if (d_Q_accum_buffer_.allocation() != nullptr) {
```

Comment:
Factor out this check (set if non-null) into a function to avoid repeating it N times in this function?

Reply:
fixed